src/ext/ext.ts

```diff
@@ -1,28 +1,191 @@
-function fetchContainerTabs(qItems: QChild[], children: PropertiesChild[]) {
+import ContainerHandler from './containerHandler';
+import containerUtil from '../utils/container-util';
+
+function fetchContainerTabs(qItems: QChild[], children: LayoutChild[], { translator, sense }: EnvironmentType) {
   const options: DropdownOption[] = [];
   children.forEach((child, index) => {
     const item = qItems.find(
       (item) => child.refId === item.qData.containerChildId || child.refId === item.qData.qExtendsId
     );
     if (item) {
-      // ToDo - create a new nebula hook that picks up the registered visualizations
-      // if (child?.label === '') {
-      //   const libInfo = Visualizations.getType(item.qData.visualization).getLibraryInfo();
-      //   translation = libInfo.translationKey ? translator.get(libInfo.translationKey) : libInfo.name;
-      // } else {
-      //   translation = child.label;
-      // }
-      options[index] = { translation: child.label, value: item.qInfo.qId };
+      const translation = containerUtil.getTranslationFromChild(
+        { ...child, ...item, visible: undefined },
+        translator,
+        sense.visualizationApi?.visualizations
+      );
+      options[index] = { translation, value: item.qInfo.qId };
     }
   });
   return options;
 }
 
-export default function ext() {
+export default function ext(env: EnvironmentType) {
+  const containerHandler = ContainerHandler(env.translator, env.sense.visualizationApi);
   const data = {
     uses: 'data',
     show: false,
   };
+  const content = {
+    type: 'items',
+    translation: 'Object.Container.Props.Content',
+    items: [
+      {
+        translation: 'Object.Container.Props.Chart',
+        component: 'text',
+        show(properties: ContainerProperties) {
+          return properties.children.length <= 1;
+        },
+      },
+      {
+        translation: 'Common.Charts',
+        component: 'text',
+        show(properties: ContainerProperties) {
+          return properties.children.length > 1;
+        },
+      },
+      {
+        type: 'array',
+        ref: 'children',
+        itemTitleRef(properties: ContainerProperties) {
+          return properties.label;
+        },
+        allowAdd: true,
+        customAdd: true,
+        allowMove: true,
+        allowRemove: true,
+        addTranslation: 'Object.Container.Props.Add',
+        grouped: true,
+        remove(item: PropertiesChild, _properties: ContainerProperties, handler: PropertyHandler, args: PropertyArgs) {
+          handler.layout.qChildList.qItems.forEach((child: QChild) => {
+            if (
+              child.qData.containerChildId === item.refId ||
+              (child.qData.qExtendsId && child.qData.qExtendsId === item.refId)
```

Comment:
If it changes. it will return true in case both `child.qData.qExtendsId` and `item.refId` are undefined. However, I picked this condition from the old code and currently I cannot see any reason why the `item.refId` will ever be undefined.
Fixing

Reply:
Fixed